src/peft/utils/save_and_load.py

```diff
@@ -345,6 +356,23 @@ def set_peft_model_state_dict(
     else:
         state_dict = peft_model_state_dict
 
+    # handle auxiliary training wrappers by getting each of them and translating saved state dict key
+    # (which does not include the adapter name) to loaded state dict key (which includes the adapter name).
+    for name, module in model.named_modules():
+        if isinstance(module, AuxiliaryTrainingWrapper):
+            for k, _ in module.adapter_state_dict(adapter_name).items():
+                # each saved state dict is adapter specific, i.e. does not contain the adapter name
+                # but the loaded state dict does include adapter names since we can have multiple.
+                k_no_adapter = k.replace(f".{adapter_name}", "")
+                store_key = f"{name}.{k}"
+                lookup_key = f"{name}.{k_no_adapter}"
+
+                state_dict[store_key] = peft_model_state_dict[lookup_key]
+
+                # delete the old key from the previous `state_dict = peft_model_state_dict` statement.
+                # TODO can be removed once the model_to_save code is refactored
```

Comment:
```suggestion
                # TODO can be removed once the modules_to_save code is refactored
```
?

Also, what would need to be refactored?

Reply:
In a perfect world I'd drop the `modules_to_save` handling above completely and just use `AuxiliaryTrainingWrapper` - that's what I meant with refactoring. Looking at it now it is, however, unlikely that the commented code will differ much. So the question is just: do we want to remove the `modules_to_save` handling and streamline it or do we want to keep it? I think it is safe to drop it but I wanted a second opinion beforehand.

Reply:
I thought about this and I don't it makes sense to keep both code pieces in there. They should do the same and if they don't it is a bug either way. If this is making trouble in other tests we can discuss working around it / excluding modules to save explicitly.